### PR TITLE
feat: Add endpoints for Interviews

### PR DESCRIPTION
**Pull request by wilsonplau**
### Description

What new features or functions does this PR provide? What bugs does it fix?
- Adds several endpoints and the repository / service to support that
- Moved auth middleware to `index.ts` as well
- Includes:
   - `GET /api/interviews`
   - `GET /api/interviews/:interviewId`
   - `POST /api/interviews`
   - `POST /api/interviews/:interviewId`

### Testing

How did you test the code in the pull request?
- Manual tests in Insomnia for now
- Adding a testing setup in a separate PR

### Checklist

- [X] I provided a description of the contents of this pull request.
- [ ] I've linked this pull request to an issue.
- [X] I've assigned myself as the Assignee and labelled the PR correctly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,12 @@
   "name": "mcitmocks",
   "version": "1.0.0",
   "scripts": {
-    "start": "ts-node index"
+    "start": "ts-node index",
```
Comment (joshmhanson) [inline package.json]: This comma is giving me an error

**Review by joshmhanson — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (wilsonplau) [reply package.json]: Fixed in 1540e13 and merged develop to fix conflicts.

**Review by yuyu0227 — APPROVED** [review]
looks good to me

File: models/InterviewRepository.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    this.prisma = prismaClient;
+  }
+
+  async getInterviewById(interviewId: string): Promise<Interview> {
+    const interview = await this.prisma.interview.findUnique({ where: { id: interviewId } });
+    if (!interview) throw new Error("Error: interviewId not found");
+    return interview;
+  }
+
+  async getInterviewsByUserId(userId: string): Promise<Interviews> {
```
Comment (yuyu0227) [inline models/InterviewRepository.ts:17]: I'm wondering is it necessary to add a "getInterviewsByUserIdAndStartTime" ?  If not, then there will be many interviews that does not belong to the desired time range when we load interviews for a specific week  (or is it better to filter interviews in frontend after fetching all interviews given a user?)

Comment (wilsonplau) [reply models/InterviewRepository.ts:17]: We can always add more endpoints when we need it.

**Pull request merged**